core/perf_test/PerfTest_Stream.cpp

```diff
@@ -0,0 +1,274 @@
+//
+//@HEADER
+
+/**
+ * @file PerfTest_Stream.cpp
+ * @brief Implementation of STREAM benchmark operations for Kokkos.
+ *
+ * @details This file provides a set of memory bandwidth benchmarks based on the
+ * STREAM benchmark suite. It implements the five core STREAM operations (Set,
+ * Copy, Scale, Add, and Triad) using Kokkos parallel primitives. It includes
+ * validation.
+ *
+ * The implementation strives to use as few Kokkos features as possible, and so
+ * validation is performed on the host rather than via parallel_reduce
+ */
+
+#include <Kokkos_Core.hpp>
+#include <benchmark/benchmark.h>
+#include "Benchmark_Context.hpp"
+
+constexpr static double aInit  = 1.0;
+constexpr static double bInit  = 2.0;
+constexpr static double cInit  = 3.0;
+constexpr static double scalar = 4.0;
+
+using StreamDeviceArray =
+    Kokkos::View<double*, Kokkos::MemoryTraits<Kokkos::Restrict>>;
+using StreamHostArray = typename StreamDeviceArray::HostMirror;
+
+// different than benchmarks/stream, which uses int
+// wide index types are common as GPU memory grows
+using StreamIndex = int64_t;
+using Policy      = Kokkos::RangePolicy<Kokkos::IndexType<StreamIndex>>;
+
+template <typename V>
+void perform_set(V& a, const double scalar_) {
+  Kokkos::parallel_for(
+      "set", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { a[i] = scalar_; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_copy(V& a, V& b) {
+  Kokkos::parallel_for(
+      "copy", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { b[i] = a[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_scale(V& b, V& c, const double scalar_) {
+  Kokkos::parallel_for(
+      "scale", Policy(0, b.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { b[i] = scalar_ * c[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_add(V& a, V& b, V& c) {
+  Kokkos::parallel_for(
+      "add", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { c[i] = a[i] + b[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+void perform_triad(V& a, V& b, V& c, const double scalar_) {
+  Kokkos::parallel_for(
+      "triad", Policy(0, a.extent(0)),
+      KOKKOS_LAMBDA(const StreamIndex i) { a[i] = b[i] + scalar_ * c[i]; });
+
+  Kokkos::fence();
+}
+
+template <typename V>
+int validate_array(V& a_dev, const double expected) {
+  auto a = Kokkos::create_mirror_view(a_dev);
+  Kokkos::deep_copy(a, a_dev);
+
+  double error = 0.0;
+  for (size_t i = 0; i < a.size(); ++i) {
+    error += std::abs(a[i] - expected);
+  }
+  double avgError = error / (double)a.size();
+
+  const double epsilon = 1.0e-13;
+  return std::abs(avgError / expected) > epsilon;
+}
+
+template <unsigned MemTraits>
+static void StreamSet(benchmark::State& state) {
+  const size_t N8                 = std::pow(state.range(0), 8);
+  static constexpr int DATA_RATIO = 1;
+
+  Kokkos::View<double*, Kokkos::MemoryTraits<MemTraits>> a("a", N8);
```

Comment:
```suggestion
  Kokkos::View<double*, Kokkos::MemoryTraits<MemTraits>> a(Kokkos::view_alloc(Kokkos::WithoutInitializing, "a"), N8);
```
(such a change could be made everywhere where you're overwriting the view immediately)